src/back/CodeGen/Expr.hs

```diff
@@ -1062,11 +1054,41 @@ instance Translatable A.Expr (State Ctx.Context (CCode Lval, CCode Stat)) where
     else
       error $ "Expr.hs: Cannot translate forward of ''" ++ show expr ++ "'"
 
-  translate A.Forward{A.forwardExpr = A.FutureChain{}} =
-    error "Expr.hs: Forwarding of chaining not implemented"
-  translate A.Forward{A.forwardExpr} =
-    error $ "Expr.hs: Target of forward is not method call or future chain: '" ++
-            show forwardExpr ++ "'"
+  translate A.Forward{A.emeta, A.forwardExpr = fchain@A.FutureChain{A.future, A.chain}} = do
+    (nfuture,tfuture) <- translate future
+    (nchain, tchain)  <- translate chain
+    eCtx <- gets $ Ctx.getExecCtx
+    withForwarding <- gets Ctx.withForwarding
+    let ty = getRuntimeType chain
+        dtraceExit = head (getDtraceExit eCtx)
+    if withForwarding
```

Comment:
To be honest, I'm not entirely sure that I'm correct. I always viewed methods based on how they were called, rather than what their contents were. This means that whether a method uses forward or not should not affect how it is called. It seems that the implementation is quite different. 

I tried to break the implementation by mixing things up with traits, to trick it into calling the non-forward version of the method when the forward version is expected, but I couldn't do it.